pingcap/tiflash
# clean up segments with large space amplification ratio generated by logical split

### What problem does this PR solve?

Issue Number: close #5817 

Problem Summary: After logical split, the result segments will ref a `DTFile` whose range is almost twice as larger than the corresponding segments and these cause space amplification problem which cannot be ignored.

### What is changed and how it works?
1. calculate the ratio of the valid_size/valid_bytes of `StableValueSpace` to the size/bytes of underlying `DTFile` and trigger a bg gc if the ratio is too low.

### Check List

Tests 

1. enable logical split and load ch-benchmark;
2. restart to check whether the store size change;
3. patch new binary to check whether the store size change;
After patch, the bg gc start to clean the segments generated by logical split and the store size begin to decrease:
![image](https://user-images.githubusercontent.com/47731263/189851521-a0e563d2-2924-43f4-aa95-20a7f38d0516.png)
![image](https://user-images.githubusercontent.com/47731263/189852456-68f1d94a-079f-4b6e-bf74-ab9dd674e312.png)


Side effects


Documentation


### Release note



```release-note
None
```
